.gitattributes

```diff
@@ -0,0 +1 @@
+* text=auto
```

Comment:
The [git documentation](https://git-scm.com/docs/gitattributes) says that:

```
Set to string value "auto"
When text is set to "auto", the path is marked for automatic end-of-line conversion. If Git decides that the content is text, its line endings are converted to LF on checkin. When the file has been committed with CRLF, no conversion is done.
```

So it is still possible for people to push files with CRLF to the repo, since the conversion is only done on pull, according to the last line? A better enforcement would be needed, because adding this only hides the problem. 😕

Reply:
I think the documentation is written quite vaguely, so you might have taken away the wrong understanding from it.

My understanding (which I believe is correct because I spent some time playing around with the settings) is that:
- The `auto` setting converts (all files that are determined by Git to be text and not binary) from CRLF to LF **_upon check in_**.
- This means that I could be committing a CRLF file, but what actually goes into the repo is the converted LF version:

<img width="585" alt="Screenshot 2019-03-19 at 1 00 24 AM" src="https://user-images.githubusercontent.com/9073504/54548249-6ce69c80-49e2-11e9-8f3b-b7a86f2c78d3.png">

- The file remains as CRLF on my local repo. I haven't tested it, but my hunch is that it will get overwritten by the actual LF version when pulling from remote later on. In any case, the source of truth (our git remote) will remain as LF, and any inconsistencies will be limited to individual clones.
- As this is a git feature, it makes sense to me that it only controls the way files are versioned in the repo, rather than modifying what's on your local machine. So I think this is all the enforcement power there is to `.gitattributes`.

The gist is, after I looked through the documentation and tested different settings, I came to the conclusion that `* text=auto` does what we want in the easiest and most pain-free way.

Reply:
Thanks for your explanation, checked it out on my side, seems to be working as you explained. :+1: